core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -184,14 +185,15 @@ Query parsePercolatorDocument(String id, BytesReference source) {
         }
     }
 
+    //norelease this method parses from xcontent to lucene query, need to re-investigate how to split context here
     private Query parseQuery(String type, XContentParser parser) {
```

Comment:
clarify the comment and add reference to the open issue? We don't need to investigate, we just have to do it when ready :)